x/airdrop/keeper/claim_test.go

```diff
@@ -0,0 +1,450 @@
+				s.Require().ErrorContains(actualError, tc.expectedError)
+				return
+			}
+			s.Require().NoError(actualError, "no error expected when claiming early")
+
+			// Check that the user was updated
+			userAllocation := s.MustGetUserAllocation(AirdropId, claimer.String())
+			s.Require().Equal(tc.expectedClaimed, userAllocation.Claimed.Int64(), "claimed")
+			s.Require().Equal(tc.expectedForfeited, userAllocation.Forfeited.Int64(), "forfeited")
+			s.Require().Equal(types.CLAIM_EARLY, userAllocation.ClaimType, "claim types")
+			for _, allocation := range userAllocation.Allocations {
+				s.Require().Zero(allocation.Int64(), "allocations should be 0")
+			}
+
+			// Confirm funds were decremented from the distributor
+			expectedDistributorBalance := initialDistributorBalance.Sub(sdkmath.NewInt(tc.expectedUserBalanceChange))
+			actualDistributorBalance := s.App.BankKeeper.GetBalance(s.Ctx, distributor, RewardDenom).Amount
+			s.Require().Equal(expectedDistributorBalance.Int64(), actualDistributorBalance.Int64(), "distributor balance")
+
+			// Confirm funds were sent to the user
+			claimerBalance := s.App.BankKeeper.GetBalance(s.Ctx, claimer, RewardDenom).Amount
+			s.Require().Equal(tc.expectedUserBalanceChange, claimerBalance.Int64(), "claimer balance")
+		})
+	}
+}
+
+func (s *KeeperTestSuite) TestLinkAddresses() {
+	testCases := []struct {
+		name                string
+		initialClaimType    types.ClaimType
+		initialClaimed      sdkmath.Int
+		strideAllocations   []int64
+		hostAllocations     []int64
+		expectedAllocations []int64
+		expectedError       string
+	}{
+		{
+			name:                "no stride allocations",
+			initialClaimType:    types.CLAIM_DAILY,
+			initialClaimed:      sdkmath.ZeroInt(),
+			strideAllocations:   nil,
+			hostAllocations:     []int64{10, 20, 30},
+			expectedAllocations: []int64{10, 20, 30},
+		},
+		{
+			name:                "stride and host allocations",
+			initialClaimType:    types.CLAIM_DAILY,
+			initialClaimed:      sdkmath.NewInt(10),
+			strideAllocations:   []int64{10, 20, 30},
+			hostAllocations:     []int64{40, 50, 60},
+			expectedAllocations: []int64{50, 70, 90},
+		},
+		{
+			name:                "user previously claimed early",
+			initialClaimType:    types.CLAIM_EARLY,
+			initialClaimed:      sdkmath.NewInt(20),
+			strideAllocations:   []int64{0, 0, 0},
+			hostAllocations:     []int64{40, 50, 60},
+			expectedAllocations: []int64{40, 50, 60},
+		},
+		{
+			name:                "link after no more left to claim",
+			initialClaimType:    types.CLAIM_EARLY,
+			initialClaimed:      sdkmath.NewInt(100),
+			strideAllocations:   []int64{0, 0, 0},
+			hostAllocations:     []int64{0, 0, 0},
+			expectedAllocations: []int64{0, 0, 0},
+		},
+		{
+			name:              "different allocation lengths",
+			initialClaimType:  types.CLAIM_DAILY,
+			initialClaimed:    sdkmath.NewInt(100),
+			strideAllocations: []int64{10, 20, 30},
+			hostAllocations:   []int64{40, 50},
```

Comment:
Similar to prev comment about lengths - how do we reach a state where the lengths aren't the same?

Reply:
shouldn't be possible! Just a sanity check